File: 2. Clean Architecture Objective-C/Configurator.xctemplate/___FILEBASENAME___Configurator.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright (c) ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

#import "___VARIABLE_sceneName:identifier___Configurator.h"
#import "___VARIABLE_sceneName:identifier___ViewController.h"
#import "___VARIABLE_sceneName:identifier___Router.h"
#import "___VARIABLE_sceneName:identifier___Interactor.h"
#import "___VARIABLE_sceneName:identifier___Presenter.h"

@implementation ___VARIABLE_sceneName:identifier___Configurator

+ (void)injectDependenciesForViewController:(___VARIABLE_sceneName:identifier___ViewController *)viewController {
    if (viewController.output != nil) {
        return;
    }
    
    ___VARIABLE_sceneName:identifier___Router *router = [___VARIABLE_sceneName:identifier___Router new];
    router.viewController = viewController;
    viewController.router = router;
    
    ___VARIABLE_sceneName:identifier___Presenter *presenter = [___VARIABLE_sceneName:identifier___Presenter new];
    ___VARIABLE_sceneName:identifier___Interactor *interactor = [___VARIABLE_sceneName:identifier___Interactor new];
    viewController.output = interactor;
    interactor.output = presenter;
    presenter.output = viewController;
}

@end

File: 1. Clean Architecture Swift/Configurator.xctemplate/___FILEBASENAME___Configurator.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright (c) ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

import UIKit

/**
    Class to build view controller's dependency tree. Makes and links ViewController, Interactor, and Presenter objects.
 */
class ___VARIABLE_sceneName:identifier___Configurator {
    static func inject(dependenciesFor viewController: ___VARIABLE_sceneName:identifier___ViewController) {
        if viewController.output != nil {
            return
        }
        
        let router = ___VARIABLE_sceneName:identifier___Router()
        router.viewController = viewController
        viewController.router = router
        
        let presenter = ___VARIABLE_sceneName:identifier___Presenter()
        let interactor = ___VARIABLE_sceneName:identifier___Interactor()
        viewController.output = interactor
        interactor.output = presenter
        presenter.output = viewController
    }
}
